File: Dockerfile.swift
FROM gocd/gocd-agent-debian-11:v23.1.0

ENV DEBIAN_FRONTEND=noninteractive

USER root

# Base packages
RUN apt-get update && \
  apt-get install -y --no-install-recommends \
  curl \
  wget \
  ca-certificates \
  make \
  binutils \
  git \
  gnupg2 \
  libc6-dev \
  libcurl4 \
  libedit2 \
  libgcc-9-dev \
  libpython2.7 \
  libsqlite3-0 \
  libstdc++-9-dev \
  libxml2 \
  libz3-dev \
  pkg-config \
  tzdata \
  zlib1g-dev \
  python3 \
  python3-pip

# swift
RUN wget https://download.swift.org/swift-5.8-release/ubuntu2004/swift-5.8-RELEASE/swift-5.8-RELEASE-ubuntu20.04.tar.gz \
    && tar -xvzf swift-5.8-RELEASE-ubuntu20.04.tar.gz \
    && mv swift-5.8-RELEASE-ubuntu20.04 /opt/swift \
    && rm -rf swift-5.8* \
    && wget -q -O - https://swift.org/keys/all-keys.asc | gpg --import - \
    && echo "export PATH=/opt/swift/usr/bin:$PATH" >> ~/.bashrc

File: Dockerfile.php
FROM gocd/gocd-agent-debian-11:v23.1.0

ENV DEBIAN_FRONTEND=noninteractive

USER root

# Base packages
RUN apt-get update && \
  apt-get install -y --no-install-recommends \
  curl \
  wget \
  gnupg \
  ca-certificates \
  make \
  lsb-base \
  lsb-release

# php repo
RUN echo "deb https://packages.sury.org/php/ $(lsb_release -sc) main" | tee /etc/apt/sources.list.d/sury-php.list
RUN wget -qO - https://packages.sury.org/php/apt.gpg | apt-key add -

RUN apt-get update && \
  apt-get install -y --no-install-recommends \
  php8.2 \
  php8.2-cli \
  php8.2-common \
  php8.2-zip

# Install Composer
RUN curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer